### DISCUSSION TITLE
Splitting files into smaller components

### DISCUSSION THREAD

**Discussion by brbucher** [CMSgov/price-transparency-guide#330] [Q&A] [answered]
Would it be possible to split the output for the in network file into "per contract" files instead of reporting every negotiated price for all contracts?

Currently a single file can contain rates from many contracts. This requires all of the rates to be gathered and stored before we are able to print them to a file.
If we could instead print a smaller file for each contract, then the overhead of having to process everything and hold on to all of the data before being able to write a file would be significantly reduced.

For an additional point of reference: mjbriggs brought this up in [discussion 278](https://github.com/CMSgov/price-transparency-guide/discussions/278#discussioncomment-1695819)
"
Within a health plan, there can be multiple prices for a billing code. These prices come from contracted rates. Contracted rates will live inside a contract between a healthcare organization and a payer … Being able to price and print per contract would yield tremendous scalability improvements.
"

**Comment by brbucher** [comment]
Bumping this thread, any thoughts @shaselton-usds?

**Comment by shaselton-usds** [comment, marked as answer]
@brbucher If you are able to produce a valid in-network file for each contract, that could be referenced in the table-of-contents file, that should be fine. Is that what you're asking?

**Reply by brbucher** [reply]
Yes it was, thanks for the clarification!
At this point we will be sticking to the full file output, but it is good to know that we have the option to swap to a file-per-contract model in the future.
